backends/hip-ref/ceed-hip-ref-vector.c

```diff
@@ -752,18 +1212,22 @@ int CeedVectorCreate_Hip(CeedSize n, CeedVector vec) {
   ierr = CeedSetBackendFunction(ceed, "Vector", vec, "HasBorrowedArrayOfType",
                                 CeedVectorHasBorrowedArrayOfType_Hip);
   CeedChkBackend(ierr);
-  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "SetArray",
-                                CeedVectorSetArray_Hip); CeedChkBackend(ierr);
-  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "TakeArray",
-                                CeedVectorTakeArray_Hip); CeedChkBackend(ierr);
+  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "CheckArrayStatus",
+                                CeedVectorCheckArrayStatus_Hip); CeedChkBackend(ierr);
+  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "SetArrayGeneric",
+                                CeedVectorSetArrayGeneric_Hip); CeedChkBackend(ierr);
+  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "TakeArrayGeneric",
+                                CeedVectorTakeArrayGeneric_Hip); CeedChkBackend(ierr);
   ierr = CeedSetBackendFunction(ceed, "Vector", vec, "SetValue",
                                 (int (*)())(CeedVectorSetValue_Hip)); CeedChkBackend(ierr);
-  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "GetArray",
-                                CeedVectorGetArray_Hip); CeedChkBackend(ierr);
-  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "GetArrayRead",
-                                CeedVectorGetArrayRead_Hip); CeedChkBackend(ierr);
-  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "GetArrayWrite",
-                                CeedVectorGetArrayWrite_Hip); CeedChkBackend(ierr);
+  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "SyncArrayGeneric",
+                                CeedVectorSyncArrayGeneric_Hip); CeedChkBackend(ierr);
```

Comment:
The `SyncArray` backend function was not "connected" previously (for any backends, not just hip-ref), so `CeedVectorSyncArray` always called the default interface implementation.  The `Sync` function in this backend was only used within the backend, called by other backend functions in this file.  Changing the `Sync` to be used from the interface call required changing the use of `NeedSync` in several functions; now the check happens in `Sync` itself.

Reply:
It might be worth pulling the commits to fix this oversight into a separate PR that we can quickly merge

Reply:
I can make that PR today